Admin/Get-MigrationReport.ps1

```diff
@@ -0,0 +1,214 @@
+
+<#
+Version: 2.0
+#  DISCLAIMER:
+# THIS CODE IS SAMPLE CODE. THESE SAMPLES ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND.
+# MICROSOFT FURTHER DISCLAIMS ALL IMPLIED WARRANTIES INCLUDING WITHOUT LIMITATION ANY IMPLIED WARRANTIES OF MERCHANTABILITY OR OF FITNESS FOR
+# A PARTICULAR PURPOSE. THE ENTIRE RISK ARISING OUT OF THE USE OR PERFORMANCE OF THE SAMPLES REMAINS WITH YOU. IN NO EVENT SHALL
+# MICROSOFT OR ITS SUPPLIERS BE LIABLE FOR ANY DAMAGES WHATSOEVER (INCLUDING, WITHOUT LIMITATION, DAMAGES FOR LOSS OF BUSINESS PROFITS,
+# BUSINESS INTERRUPTION, LOSS OF BUSINESS INFORMATION, OR OTHER PECUNIARY LOSS) ARISING OUT OF THE USE OF OR INABILITY TO USE THE
+# SAMPLES, EVEN IF MICROSOFT HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES. BECAUSE SOME STATES DO NOT ALLOW THE EXCLUSION OR LIMITATION
+# OF LIABILITY FOR CONSEQUENTIAL OR INCIDENTAL DAMAGES, THE ABOVE LIMITATION MAY NOT APPLY TO YOU.
+
+
+.SYNOPSIS
+This script generates all the needed reports to troubleshoot a move request from or to Exchange Online/OnPrem. It can handle multiple mailboxes at once.
+
+.DESCRIPTION
+This is a PowerShell script that is used for generating reports related to mailbox migrations in Microsoft Exchange. The script accepts a mandatory parameter called $Identity, which should be an array of mailbox names. The script exports various types of reports as XML files, including:
+- MoveRequest: A report containing information about the move request for the specified mailbox.
+- MoveRequestStatistics: A report containing statistical information about the move request, including details about the number of items that were moved, failed, or are in a warning state.
+- UserMigration
+- UserMigrationStatistics
+- MigrationBatch
+- MigrationEndPoint
+- MigrationConfig
+The script also exports a report containing statistics for the specified mailbox, as well as a report containing the move history for the specified mailbox. Finally, the script logs any errors that occur during the export process to a log file called LogFile.txt.
+
+
+.NOTES
+    This script should be excuted in exchange online or Exchange OnPrem Powershell module.
+.EXAMPLE
+    .\Get-MigrationReports -Identity Mustafa@contoso.com
+    .\Get-MigrationReports -Identity user1@contoso.com, user2@contoso.com, user3@contoso.com
+
+# By Mustafa Nassar, Use at your own risk.  No warranties are given.
+
+#>
+
+[CmdletBinding()]
+param (
+    [Parameter( Mandatory = $true, HelpMessage = 'You must specify the name of a mailbox or mailboxes:')] [array] $Identity,
+    [Parameter(Mandatory = $false, HelpMessage = 'Specify the folder where the reports should be saved')] [string] $OutputFolder
+)
+
+# Set the default output folder if none is specified
+if (-not $OutputFolder) {
+    $OutputFolder = 'Get-MigrationReports'
+}
+# Set the log file path
+$logFile = "$OutputFolder\LogFile.txt"
+
+function Export-XMLReports {
+    # Export XML reports:
+    try {
+        if (-not $null -eq $MoveRequest) {
```

Comment:
For all ones similar to this instead of `-not $null -eq $Variable` do `$null -ne $Variable`